### PR TITLE
Update binder to use conda, which allows us to install our own nodejs.

### PR DESCRIPTION
**Pull request by jasongrout**
## Code changes

Our binder doesn’t work right now because the platform has node 10, but we now require node 12. This updates our binder to install things with conda, which allows us to update node.

### CODE REVIEW

**Comment by jasongrout** [comment]
The build seemed to be okay, but I get a 500 server error when using the binder link above. Not sure what the problem is.

File: binder/environment.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+dependencies:
+  - jupyter-server-proxy
+  - matplotlib-base
+  - nodejs
+  - numpy
+  - pip
+  - python
+  - xeus-python
+  - pip:
+    - vega-datasets
```
Comment (jtpio) [inline binder/environment.yml:13]: It looks like `vega-datasets` is also available on conda forge: https://anaconda.org/conda-forge/vega_datasets

```suggestion
  - vega_datasets
  - xeus-python
```

**Comment by jtpio** [comment]
> The build seemed to be okay, but I get a 500 server error when using the binder link above. Not sure what the problem is.

Maybe it's possible to check what is happening locally with something like this:

```
docker run -v /var/run/docker.sock:/var/run/docker.sock jupyter/repo2docker:0.11.0-184.g30ef220 jupyter-repo2docker --user-name jovyan --user-id 1000 https://github.com/jasongrout/jupyterlab --ref binder
```

Using the latest docker image for `repo2docker` since the version published on PyPI is not the one used on mybinder.org: https://github.com/jupyterhub/mybinder.org-deploy/blob/1bb53c3dea862c6cf18bd0b09a77007837631297/mybinder/values.yaml#L83

**Comment by jasongrout** [comment]
The binder link seems to work now!

**Review by jtpio — APPROVED** [review]
Nice, thanks!

**Comment by jtpio** [comment]
For reference, there is an open PR to update node to a newer version in `repo2docker` (still wip at the moment): https://github.com/jupyterhub/repo2docker/pull/983

**Pull request merged**